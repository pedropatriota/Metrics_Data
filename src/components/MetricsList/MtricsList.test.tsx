import "@testing-library/jest-dom";

import { render } from "@testing-library/react";

import { MetricsList } from ".";

const columns = {
  id: true,
  label: true,
  value: true,
  type: true,
  description: true,
  category: true,
};

const metrics = [
  {
    id: "oee",
    label: "oee",
    value: 0.68,
    type: "percentage",
    description: "The overall equipment efficiency in %",
    category: "efficiency",
  },
];

describe("Snapshot test", () => {
  it("Should  represent the component", () => {
    // eslint-disable-next-line testing-library/render-result-naming-convention
    const metricsList = render(
      <MetricsList
        columns={columns}
        metrics={metrics}
        expanded={0}
        toggleAccordion={() => console.log("toggled")}
      />
    );

    expect(metricsList).toMatchSnapshot();
  });
});
