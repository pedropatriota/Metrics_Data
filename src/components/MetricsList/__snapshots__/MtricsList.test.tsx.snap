// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test Should  represent the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdvvtL ezMjmi"
      >
        <header>
          <strong>
            Metrics Table (
            1 metrics
            )
          </strong>
        </header>
        <table>
          <thead>
            <tr>
              <th>
                ID
              </th>
              <th>
                Label
              </th>
              <th>
                Type
              </th>
              <th>
                Value
              </th>
              <th>
                Category
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                oee
              </td>
              <td>
                oee
              </td>
              <td>
                percentage
              </td>
              <td>
                0.68
              </td>
              <td>
                efficiency
              </td>
            </tr>
            <tr
              class="sc-gsDKAQ iakgtY"
            >
              <td>
                <strong>
                  Description:
                </strong>
                 
                The overall equipment efficiency in %
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdvvtL ezMjmi"
    >
      <header>
        <strong>
          Metrics Table (
          1 metrics
          )
        </strong>
      </header>
      <table>
        <thead>
          <tr>
            <th>
              ID
            </th>
            <th>
              Label
            </th>
            <th>
              Type
            </th>
            <th>
              Value
            </th>
            <th>
              Category
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              oee
            </td>
            <td>
              oee
            </td>
            <td>
              percentage
            </td>
            <td>
              0.68
            </td>
            <td>
              efficiency
            </td>
          </tr>
          <tr
            class="sc-gsDKAQ iakgtY"
          >
            <td>
              <strong>
                Description:
              </strong>
               
              The overall equipment efficiency in %
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
